src/pages/modifyManage/ModifyManage.tsx

```diff
@@ -35,138 +39,203 @@ import * as S from "./ModifyManage.styled";
+      }));
     }
-  }, [data]);
+  }, [data, isSuccess]);
 
-  const { mutateAsync: updatePerformance } = useUpdatePerformance();
+  useEffect(() => {
+    const pageTitle =
+      modifyState.ModifyManageStep === 1
+        ? "공연 수정하기"
+        : modifyState.ModifyManageStep === 2
+          ? "공연 수정하기"
+          : "미리보기";
+    setHeader({
+      headerStyle: NAVIGATION_STATE.ICON_TITLE_SUB_TEXT,
+      title: pageTitle,
+      subText: "삭제",
+      leftOnClick: handleLeftBtn,
+      rightOnClick: handleRightBtn,
+    });
+  }, [modifyState.ModifyManageStep, modifyState.isBookerExist]);
+
+  const handleInputChange = (field: keyof State, value: State[keyof State]) => {
+    dispatch({ type: "SET_FIELD", field, value });
+  };
+
+  const handleModifyManageStep = () => {
+    setModifyState((prev) => ({ ...prev, ModifyManageStep: prev.ModifyManageStep + 1 }));
+  };
+
+  const handleModifyState = (field: keyof ModifyState, value: ModifyState[keyof ModifyState]) => {
+    setModifyState((prev) => ({ ...prev, [field]: value })); //브래킷 표기법 필수
+  };
 
+  //비즈니스 로직 분리 - 공연 수정하기 PUT 요청
   const handleComplete = async () => {
-    const filteredCastList = castList.filter(
+    const filteredCastList = dataState.castList.filter(
       (cast) => cast.castName || cast.castRole || cast.castPhoto
     );
-    const filteredStaffList = staffList.filter(
+    const filteredStaffList = dataState.staffList.filter(
       (staff) => staff.staffName || staff.staffRole || staff.staffPhoto
     );
 
-    const formData = {
-      performanceId: Number(performanceId),
-      performanceTitle,
-      genre: genre as "BAND" | "DANCE" | "PLAY" | "ETC",
-      runningTime,
-      performanceDescription,
-      performanceAttentionNote,
-      accountNumber,
-      posterImage,
-      performanceTeamName,
-      performanceVenue,
-      performancePeriod,
-      performanceContact,
-      ticketPrice,
-      totalScheduleCount,
-      scheduleList,
-      castList: filteredCastList,
-      staffList: filteredStaffList,
-      bankName: (!!bankName ? bankName : "NONE") as BANK_TYPE,
-      isBookerExist,
-      accountHolder,
-    };
-
     try {
-      const res = await updatePerformance(formData);
-
-      if (res?.status === 200) {
-        openAlert({
-          title: "공연 수정이 완료됐어요.",
-          subTitle: "변경된 사항(시간, 장소 등)은 예매자에게\n 개별적으로 반드시 연락해주세요.",
-          okText: "네, 알겠어요",
-          okCallback: () => {
-            navigate("/gig-manage");
-          },
-        });
-      }
+      const res = await updatePerformance({
+        performanceId: Number(performanceId),
+        ...dataState,
+        castList: filteredCastList,
+        staffList: filteredStaffList,
+      });
+
+      console.log(res);
+
+      openAlert({
+        title: "공연 수정이 완료됐어요.",
+        subTitle: "변경된 사항(시간, 장소 등)은 예매자에게\n 개별적으로 반드시 연락해주세요.",
+        okText: "네, 알겠어요",
+        okCallback: () => {
+          navigate("/gig-manage");
+        },
+      });
```

Comment:
p3) 이 부분 공연 수정이 성공되었을 경우에만 alert 띄어주기 위해서, status가 200일 경우에만 동작하도록 수정해주세요! 
```suggestion
      if(res.status === 200) {
        openAlert({
          title: "공연 수정이 완료됐어요.",
          subTitle: "변경된 사항(시간, 장소 등)은 예매자에게\n 개별적으로 반드시 연락해주세요.",
          okText: "네, 알겠어요",
          okCallback: () => {
            navigate("/gig-manage");
          },
        }});
```

Reply:
엇 전 어차피 res.status가 2xx의 값(성공 응답)이라면 try 블록의 코드가 실행되고, 그 이외의 status 값이라면 catch 블록으로 넘어가는 구조니까 굳이 if(res.status === 200) 가 필요없을거라 생각해서 제거해주긴 했었습니다 !!

근데 제가 고려 못한 부분이 있을 수도 있을 것 같아서.. 혹시 if(res.status === 200) 가 필요한 이유가 뭐라고 생각하시나요? 정확히 status가 200이 아닌 경우에는 alert를 띄우지 않아야 하는 이유가 있나요?

Reply:
음 생각해보니 그렇네용! 크크 확인했습니다!